lib/util.js

```diff
@@ -308,11 +311,24 @@ function formatValue(ctx, value, recurseTimes) {
       braces = ['{', '}'];
       formatter = formatPromise;
     } else {
-      if (constructor === Object)
-        constructor = null;
-      braces = ['{', '}'];
-      empty = true;  // No other data than keys.
-      formatter = formatObject;
+      const name = Object.prototype.toString.call(value);
```

Comment:
ah thats right. Do you think I should add back the ability to use `v8::Value::IsMapIterator`?